python/ml4ir/applications/ranking/tests/test_metrics.py

```diff
@@ -232,4 +306,50 @@ def test_generate_stat_sig_based_metrics(self):
         stat_sig = metrics_helper.generate_stat_sig_based_metrics(df, metric, str(group_keys), metrics_dict)
         assert metrics_dict["stat_sig_" + metric + "_improved_groups"] == 1
         assert metrics_dict["stat_sig_" + metric + "_degraded_groups"] == 1
-        assert metrics_dict["stat_sig_" + metric + "_group_improv_perc"] == 0
\ No newline at end of file
+        assert metrics_dict["stat_sig_" + metric + "_group_improv_perc"] == 0
+
+    def test_add_top_graded_relevance_record_column(self):
+        # Create a sample DataFrame
+        data = {
+            'query_id': [1, 1, 2, 2, 3, 4, 4],
+            'relevance_score': [0.8, 0.5, 0.6, 0.9, 0.7, 0, 0]
+        }
+        df = pd.DataFrame(data)
+
+        # Call the function
+        new_col_name = 'top_relevance'
+        df_with_new_column = metrics_helper.add_top_graded_relevance_record_column(df, 'relevance_score', new_col_name)
+
+        # Verify the new column is added
+        self.assertTrue(new_col_name in df_with_new_column.columns)
+
+        # Verify the values in the new column
+        expected_values = np.array([1.0, 0.0, 0.0, 1.0, 1.0])
```

Comment:
@mohazahran I see that you return all equal top scores as clicked. How does this affect MRR computation? I worry it just ignores queries with more than 1 click. Can you update the implementation to use the record with highest "new_rank"?

Reply:
no that's not what's happening here. we have only one "clicked" record per query. not multiple.

Reply:
Why does expected_values have only 5 values and not 6?

Reply:
It's because the last query in test has no relevance signal. Both records have graded relevance = 0

Reply:
I see what is happening now. You should create a separate test named something that indicates removal of queries without total graded relevance. There's a lot of things being tested here.

Reply:
The other issue I'd like to see a test for is when there are ties in the score. I noticed `idxmax()` picks last by default. We should pick the one with the highest new ranking score IMO as essentially all the tied values are "clicks". @mohazahran

Reply:
```
In [49]: pd.Series([1, 1, 2, 3, 4]).idxmax()
Out[49]: 4
```

Reply:
MRR is generally computed on the top most click when there are multiple.

Reply:
It's because the last query in test has no relevance signal. Both records have graded relevance = 0, so it's removed